seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -405,32 +477,144 @@ def test_create_project_from_workspace(self, mock_tempfile, mock_mv_file, mock_s
+        self.assertEqual(response.json()['error'], 'Project matching query does not exist.')
+
+        # Test missing required fields in the request body
+        url = reverse(add_workspace_data, args=[PROJECT1_GUID])
+        self.client.force_login(self.manager_user)
+
+        self._test_missing_fields_in_request(url, ['uploadedFileId', 'fullDataPath', 'vcfSamples'], TEST_WORKSPACE_NAME)
+
+        self._test_sample_data_error(url)
+
+        self._test_missing_samples(url)
+
+        # Test missing loaded samples
+        self.mock_load_file.return_value = LOAD_SAMPLE_DATA
+        response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY))
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            response.json()['error'],
+            'In order to add new data to this project, new samples must be joint called in a single VCF with all previously'
+            ' loaded samples. The following samples were previously loaded in this project but are missing from the VCF:'
+            ' HG00731, HG00732, HG00733, NA19675_1, NA20870, NA20874')
+
+        # Test a valid operation
+        response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY_ADD_DATA))
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertEqual(sorted(response_json.keys()), ['familiesByGuid', 'familyNotesByGuid', 'individualsByGuid'])
+        self.assertEqual(sorted(response_json['individualsByGuid'].keys()), ['I0000019_hg00735', 'I000001_na19675', 'I000002_na19678'])
+        self.assertEqual(sorted(response_json['familiesByGuid'].keys()), ['F000001_1', 'F000015_21'])
+        self.assertEqual(sorted(response_json['familyNotesByGuid'].keys()), ['FAN000004_21_c_a_new_family'])
+        self.mock_api_logger.error.assert_not_called()
+
+        self.mock_tempfile.assert_called_with(mode='wb', delete=False)
+        self.mock_tempfile.return_value.__enter__.return_value.write.assert_called_with(
+            b's\nHG00731\nHG00732\nHG00733\nHG00735\nNA19675\nNA19675_1\nNA19678\nNA19678\nNA20870\nNA20874')
+        self.mock_mv_file.assert_called_with(
+            TEMP_PATH,
+            'gs://seqr-datasets/v02/GRCh37/AnVIL_WES/{guid}/base/{guid}_ids.txt'.format(guid=PROJECT1_GUID),
+            user=self.manager_user
+        )
+
+        self._assert_dag_trigger_requests(ADD_DATA_UPDATED_ANVIL_VARIABLES, 10, PROJECT1_GUID)
+
+        self._assert_sent_slack_message(Project.objects.get(guid=PROJECT1_GUID), "GRCh37")
+
+        url = reverse(add_workspace_data, args=[PROJECT2_GUID])
+        self._test_mv_file_and_triggering_dag_exception(url, {'guid': PROJECT2_GUID}, PROJECT2_SAMPLES, 'GRCh37', REQUEST_BODY_ADD_DATA2)
+
+        # Test logged in locally
+        remove_token(self.manager_user)  # The user will look like having logged in locally after the access token is removed
```

Comment:
this seems to be copy pasted from the above test, so it should be a helper function

Reply:
We don't support logging in locally under Anvil mode. So I'm going to remove these tests.